agent/src/main/java/com/microsoft/applicationinsights/agent/internal/config/DataOfConfigurationForException.java

```diff
@@ -28,16 +28,25 @@
  */
 public final class DataOfConfigurationForException {
     private boolean enabled = false;
-    private int stackSize = Integer.MAX_VALUE;
+    private Integer maxStackSize;
```

Comment:
why do we need Class type `Integer` instead of primitive type `int`?

Reply:
The idea is to avoid the duplication of decision logic for default values (in xml, sb and agent config). It seems more convenient to keep it in one place (see ExceptionTelemetryOptions.java). 
Null value, in case of the absence of specific configuration, seems like a natural choice.